src/test/java/org/isf/patvac/test/Tests.java

```diff
@@ -215,6 +221,305 @@ public void testListenerShouldUpdatePatientToMergedWhenPatientMergedEventArrive(
 		assertThat(result.getPatient().getCode()).isEqualTo(mergedPatient.getCode());
 	}
 
+	@Test
+	public void testMgrGetPatientVaccineToday() throws Exception {
+		VaccineType vaccineType = testVaccineType.setup(false);
+		Vaccine vaccine = testVaccine.setup(vaccineType, false);
+		Patient patient = testPatient.setup(false);
+		PatientVaccine patientVaccine = testPatientVaccine.setup(patient, vaccine, true);
+
+		patientVaccine.setVaccineDate(new GregorianCalendar());
+
+		vaccineTypeIoOperationRepository.saveAndFlush(vaccineType);
+		vaccineIoOperationRepository.saveAndFlush(vaccine);
+		patientIoOperationRepository.saveAndFlush(patient);
+		patVacIoOperationRepository.saveAndFlush(patientVaccine);
+
+		ArrayList<PatientVaccine> patientVaccines = patVacManager.getPatientVaccine(false);
+		assertThat(patientVaccines.get(patientVaccines.size() - 1).getPatName()).isEqualTo(patientVaccine.getPatName());
+	}
+
+	@Test
+	public void testMgrGetPatientVaccineLastWeek() throws Exception {
+		VaccineType vaccineType = testVaccineType.setup(false);
+		Vaccine vaccine = testVaccine.setup(vaccineType, false);
+		Patient patient = testPatient.setup(false);
+		PatientVaccine patientVaccine = testPatientVaccine.setup(patient, vaccine, true);
+
+		GregorianCalendar date = new GregorianCalendar();
+		date.add(Calendar.DAY_OF_MONTH, -3);
+		patientVaccine.setVaccineDate(date);
+
+		vaccineTypeIoOperationRepository.saveAndFlush(vaccineType);
+		vaccineIoOperationRepository.saveAndFlush(vaccine);
+		patientIoOperationRepository.saveAndFlush(patient);
+		patVacIoOperationRepository.saveAndFlush(patientVaccine);
+
+		ArrayList<PatientVaccine> patientVaccines = patVacManager.getPatientVaccine(true);
+		assertThat(patientVaccines.get(patientVaccines.size() - 1).getPatName()).isEqualTo(patientVaccine.getPatName());
+	}
```

Comment:
Same here, could it be better to have one object inside and an other one outside the range to be retrieved?

Reply:
Good idea; added second object.